File: src/jk_5/nailed/map/gameloop/instructions/InstructionTrigger.java
package jk_5.nailed.map.gameloop.instructions;

import jk_5.nailed.Nailed;
import jk_5.nailed.map.gameloop.GameThread;
import jk_5.nailed.map.gameloop.IInstruction;

/**
 * TODO: Edit description
 *
 * @author jk-5
 */
public class InstructionTrigger implements IInstruction {

    private String stat;

    @Override
    public void injectArguments(String arguments) {
        this.stat = arguments;
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        Nailed.statManager.triggerStat(this.stat);
    }
}

File: src/jk_5/nailed/map/gameloop/instructions/InstructionTrigger.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.Nailed
import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction
import jk_5.nailed.map.stats.StatManager

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionTrigger extends IInstruction {
  private var stat: String = null

  def injectArguments(arguments: String) = this.stat = arguments
  def execute(controller: GameThread) = StatManager.triggerStat(this.stat)
}
